podpac/core/data/types.py

```diff
@@ -672,13 +637,15 @@ class H5PY(DatasetCoordinatedMixin, DataSource):
         Default is 'r'. The mode used to open the HDF5 file. Options are r, r+, w, w- or x, a (see h5py.File).
     """
 
-    source = tl.Unicode()
-    dataset = tl.Any(allow_none=True)
+    source = tl.Unicode(read_only=True)
+    dataset = tl.Any(read_only=True)
+    file_mode = tl.Unicode(default_value="r", read_only=True)
```

Comment:
Do you need a set_trait in the init for this as well?

Reply:
Hmmm. I'm surprised I made `file_mode` read_only, I don't think I meant to.